### PR TITLE
Refactor and features

### PR DESCRIPTION
**Pull request by arturhasparian**
Refactor package. Add GlobalExceptionHandler. Remove builder from userDAO. Change SignUpRequest to record. Add phone number. Add Swagger docs. Remove validationHandler.

### CODE REVIEW

File: src/main/java/com/psychology/product/util/ResponseHandler.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.psychology.product.util;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Map;
+
+public class ResponseHandler {
+    public static ResponseEntity<Object> generateResponse(String messege, HttpStatus status) {
+        Map<String, Object> map = Map.of(
```
Comment (nemeritskyy) [inline src/main/java/com/psychology/product/util/ResponseHandler.java:10]: Be better to clearly indicate the type

File: src/main/java/com/psychology/product/util/exception/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import com.psychology.product.util.ResponseHandler;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+    @ExceptionHandler(BadRequestException.class)
+    public ResponseEntity<Object> badRequest(BadRequestException e) {
```
Comment (nemeritskyy) [inline src/main/java/com/psychology/product/util/exception/GlobalExceptionHandler.java]: Check return type, it different for both methods

File: src/main/java/com/psychology/product/util/ResponseHandler.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        );
+        return new ResponseEntity<>(map, status);
+    }
+
+    public static ResponseEntity<Object> generateError(String message, HttpStatus status) {
+        Map<String, Object> map = Map.of(
+                "message", message,
+                "status", status.value()
+        );
+        return new ResponseEntity<>(map, status);
```
Comment (nemeritskyy) [inline src/main/java/com/psychology/product/util/ResponseHandler.java:22]: Why we return twice status, in response and in the body?

**Review by nemeritskyy — CHANGES_REQUESTED** [review]
We can discuss unclear points

File: src/main/java/com/psychology/product/util/ResponseHandler.java (hunk 1 above)
Comment (arturhasparian) [reply src/main/java/com/psychology/product/util/ResponseHandler.java:10]: We use it in all project. If we add specific type need rewrite this method. But we try to don't repeat youself

File: src/main/java/com/psychology/product/util/ResponseHandler.java (hunk 3 above)
Comment (arturhasparian) [reply src/main/java/com/psychology/product/util/ResponseHandler.java:22]: We get status.value then return status

**Review by nemeritskyy — APPROVED** [review]
Thanks for the explanations

**Pull request merged**